nix/package.nix

```diff
@@ -38,19 +42,44 @@ writeShellScriptBin "agenix" ''
     exit 1
   }
 
-  case "$1" in
-    "help"|"--help"|"-help"|"-h")
-      show_help
-      exit 1
-      ;;
-
-    ${lib.concatStringsSep "|" allApps})
-      APP=$1
-      shift
-      echo "Collecting information about hosts. This may take a while..."
-      exec nix run .#agenix-rekey.apps.${lib.escapeShellArg stdenv.hostPlatform.system}."$APP" -- "$@"
-      ;;
-
-    *) die "Unknown command: $1" ;;
-  esac
+  APP=""
+  SHOW_TRACE_ARG=""
+  # Various Bash versions treat empty arrays as unset, which then trigger
+  # unbound variable errors.
+  PASS_THRU_ARGS=("")
```

Comment:
This should be empty `=()` on start

Reply:
That's what I started with, but Bash treats empty arrays as unset, which then errors on the first use due to `set -u` above.  I couldn't think of a better way to work around it, and the empty string had no impact in my testing.  Is there something I can clarify better with the comment above? :)

Reply:
No, only if you use `$ARR` which is wrong anyway. If you correctly expand the array with "${ARR[@]}" it will work with empty arrays and actually expand to nothing

Reply:
Thanks!  I wish that had turned up in my research on the topic.  Seems like such a simple thing to mention in the context of unbound errors with empty arrays.